lib/query/querybuilder.js

```diff
@@ -124,6 +124,20 @@ class Builder extends EventEmitter {
     return this.withWrapped(alias, statementOrColumnList, nothingOrStatement);
   }
 
+  withMaterialized(alias, statementOrColumnList, nothingOrStatement) {
```

Comment:
You implement the functionnality for all database, but only available PG and Sqlite3, so no need here.

Reply:
Do you imply this should be added to PG dialect only?

Reply:
it should throw an error on dialects which do not support it. so base method should throw, pg and sqlite should override it to actually do something useful